webportal/webapp/src/app/main/register/register.component.ts

```diff
@@ -1,17 +1,30 @@
 import { Component, OnInit } from '@angular/core';
 import {CognitoService} from '../../../services/cognito.service';
+import { environment } from '../../../environments/environment';
 
 @Component({
   selector: 'app-register',
   templateUrl: './register.component.html',
   styleUrls: ['./register.component.css']
 })
 export class RegisterComponent implements OnInit {
+  email: string;
 
   constructor(private cognitoService: CognitoService) { }
   ngOnInit() {
   }
   userLogin() {
-      this.cognitoService.login(false);
+    if (this.email.endsWith('dot.gov')) {
+      this.cognitoService.login(false); // Route to ADFS login
```

Comment:
Is it possible to make this a named parameter? e.g. `loggedIn: false`

Reply:
stackoverflow says no: https://stackoverflow.com/questions/14931824/typescript-named-parameters-like-used-in-angularjs

Reply:
theres actually a little bit of a hubbub about it: https://github.com/Microsoft/TypeScript/issues/467

Reply:
but the short answer is no